content/backend/graphql-scala/4-deferred_resolvers.md

```diff
@@ -0,0 +1,168 @@
+---
+title: Deferred Resolvers
+pageTitle: "Usage of Deferred Resolvers for better performance"
+description: "In this chapter, at first you will learn how to use deferred resolvers for better performance."
+question: "Check the correct statement about Deferred Resolvers"
+answers: ["Fetcher is higher level API for Deferred Resolver", "Deferred Resolver is of optimizing a query, Fetcher is for taking data from data source in batch","Fetcher is light version of Deferred Resolver without cache or support for relations", "Deferred Resolver and Fetcher are completely different, not related one to each other things" ]
+correctAnswer: 0
+---
+
+
+### Deferred Resolvers and Fetchers
+
+`Fetcher`s and `Deferred Resolver`s are mechanisms for batch retrieval of object from their sources like database or external API. `Deferred Resolver` provides low-level efficient API but, on the other hand, it’s more complicated in use and less secured. `Fetcher` is a specialized version of `Deferred Resolver`. It provides high-level API, it’s easier to use and usually provides all you need. It optimizes resolution of fetched entities based on its ID or relation, it deduplicates entities and caches the results, to name but a few possibilities.
```

Comment:
`low-level efficient` -> `efficient low-level`
`to name but a few possibilities` -> `and much more`?